### PR TITLE
Refactor of websocket code

### PR DESCRIPTION
**Pull request by mtps**
* Create generic WsAdapter / DecoderAdapter to abstract out the underlying implementation for json and websockets.
* Create `nodeWebSocket()` function to allow more flexibility with what can be returned from the node.
* Lots of docs.
* General cleanup (unused if checks, null checks; durations use proper non-deprecated methods, etc).
* Rename core websocket types to follow generic behavior better.

### CODE REVIEW

**Review by jphorec — APPROVED** [review]

File: es-cli/src/main/kotlin/EventsMain.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import io.provenance.eventstream.net.okHttpNetAdapter
+import io.provenance.eventstream.stream.nodeEventStream
+import io.provenance.eventstream.stream.rpc.response.MessageType
+import io.provenance.eventstream.stream.toLiveMetaDataStream
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.runBlocking
+
+fun main() = runBlocking {
+    val okHttp = defaultOkHttpClient()
+    val netAdapter = okHttpNetAdapter("ws://rpc.test.provenance.io:26657", okHttp)
```
Comment (rchaing-figure) [inline es-cli/src/main/kotlin/EventsMain.kt:15]: assume this is just a sample main? probably ok to leave the hard coded url here?

File: es-core/src/main/kotlin/io/provenance/eventstream/flow/Extensions.kt [hunk 2]
```diff
@@ -198,7 +199,7 @@ fun <T, R> Flow<T>.windowed(
         val toSkip: Int = max(step - size, 0)
         var skipped: Int = toSkip
         var lastEmittedAt: Instant? = null
-        val pollInterval = Duration.seconds(2)
+        val pollInterval = 2.seconds
```
Comment (rchaing-figure) [inline es-core/src/main/kotlin/io/provenance/eventstream/flow/Extensions.kt:202]: should this be configurable?

**Review by rchaing-figure — APPROVED** [review]
lgtm, couple questions

File: es-core/src/main/kotlin/io/provenance/eventstream/flow/Extensions.kt (hunk 2 above)
Comment (mtps) [reply es-core/src/main/kotlin/io/provenance/eventstream/flow/Extensions.kt:202]: Probably, didn't realize it was hardcoded with magic number (pre-existing).

File: es-cli/src/main/kotlin/EventsMain.kt (hunk 1 above)
Comment (mtps) [reply es-cli/src/main/kotlin/EventsMain.kt:15]: will flip to localhost (to match other main method).

**Review by rchaing-figure — APPROVED** [review]

**Pull request merged**